ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7591,7 +7591,7 @@ int ZEND_FASTCALL zend_jit_trace_hot_root(zend_execute_data *execute_data, const
 	zend_jit_op_array_trace_extension *jit_extension;
 	size_t offset;
 	uint32_t trace_num;
-	zend_jit_trace_rec trace_buffer[ZEND_JIT_TRACE_MAX_LENGTH];
+    zend_jit_trace_rec *trace_buffer = (zend_jit_trace_rec *)malloc(JIT_G(max_trace_length) * sizeof(zend_jit_trace_rec));
```

Comment:
Keep the preallocated trace_buffer. Your patch limits the upper bound anyway.
Note: In case you swittch to ``malloc()`` you need to ``free()`` the buffer later.
Anyway, it's easier and more efficient to keep the preallocated buffer.